### PR TITLE
respect special characters in filenames during request processing

### PR DESCRIPTION
**Pull request by boajer**

### CODE REVIEW

File: views/js/assets/strategies.js [hunk 1]
```diff
@@ -44,10 +44,11 @@ define([
 
                     //is slashcat we manage slash concact
                     if(data.slashcat === true){
-                        return data.baseUrl.replace(/\/$/, '') + '/' + url.toString().replace(/^\.\//, '').replace(/^\//, '');
+                        return data.baseUrl.replace(/\/$/, '') + '/' + url.directory.replace(/^\.\//, '').replace(/^\//, '')
```
Comment (krampstudio) [inline views/js/assets/strategies.js]: if you want to concat URLs we have introduced `util/url.build`

**Comment by krampstudio** [comment]
Could you update the unit test in order to reflect your changes

File: actions/class.ItemPreview.php [hunk 2]
```diff
@@ -70,10 +70,10 @@ public function render(){
         
         $uri = base64_decode($codedUri);
         $item = new core_kernel_classes_Resource($uri);
-        if($path == 'index'){
+        if($path === 'index'){
```
Comment (krampstudio) [inline actions/class.ItemPreview.php]: Do you really need it ?

Comment (boajer) [inline actions/class.ItemPreview.php]: No, really not. According http://www.phpbench.com/ and other benchmarks  strict comparison in if operator provide slightly better performance without any side affect on logic in current case. If you prefer  not to pollute commits with so tiny changes not related to essence of issue just let me know and i will not do such thing in future.

**Pull request merged**